test/augmentation/test_augmentation.py

```diff
@@ -24,6 +25,787 @@
 )
 
 
+from kornia.testing import BaseTester, default_with_one_parameter_changed, cartesian_product_of_parameters
+from kornia.augmentation.base import AugmentationBase2D
+
+# TODO same_on_batch tests?
+
+
+class CommonTests(BaseTester):
+
+    ############################################################################################################
+    # Attributes variables to set
+    ############################################################################################################
+    _augmentation_cls: Optional[Type[AugmentationBase2D]] = None
+    _default_param_set: Dict["str", Any] = {}
+    ############################################################################################################
+    # Fixtures
+    ############################################################################################################
+
+    @pytest.fixture(scope="class")
+    def param_set(self, request):
+        raise NotImplementedError("param_set must be overriden in subclasses")
+
+    ############################################################################################################
+    # Test cases
+    ############################################################################################################
+    def test_smoke(self, param_set):
+        self._test_smoke_implementation(params=param_set)
+        self._test_smoke_call_implementation(params=param_set)
+        self._test_smoke_return_transform_implementation(params=param_set)
+
+    @pytest.mark.parametrize("input_shape,expected_output_shape",
+                             [((4, 5), (1, 1, 4, 5)), ((3, 4, 5), (1, 3, 4, 5)), ((2, 3, 4, 5), (2, 3, 4, 5))])
```

Comment:
@FGeri how would you do here with random grayscale or any future augmentation that modifies the number of channels ?

Reply:
Simply by overriding the test and reparametrizing it.